### PR TITLE
Consistent capitalization of section headers

### PR DESCRIPTION
**Pull request by timhoffm**
## PR Summary

Only use a capital letter for the first word (and special names).

### CODE REVIEW

File: doc/devel/release_guide.rst [hunk 1]
```diff
@@ -94,14 +94,14 @@ Merge ``*-doc`` branch
 Merge the most recent 'doc' branch (e.g., ``v3.2.0-doc``) into the branch you
 are going to tag on and delete the doc branch on GitHub.
 
-Update supported versions in Security Policy
+Update supported versions in security policy
```
Comment (QuLogic) [inline doc/devel/release_guide.rst:97]: Security Policy is a proper noun; see other usage.

File: doc/users/github_stats.rst [hunk 2]
```diff
@@ -1,9 +1,9 @@
 .. _github-stats:
 
-GitHub Stats
-============
+GitHub statistics
+=================
 
-GitHub stats for 2021/03/31 - 2021/05/07 (tag: v3.4.1)
+GitHub statistics for 2021/03/31 - 2021/05/07 (tag: v3.4.1)
```
Comment (QuLogic) [inline doc/users/github_stats.rst:6]: This file is autogenerated, and you haven't changed the generator.

File: doc/users/license.rst [hunk 3]
```diff
@@ -12,7 +12,7 @@ licenses. Non-BSD compatible licenses (e.g., LGPL) are acceptable in
 matplotlib toolkits.  For a discussion of the motivations behind the
 licencing choice, see :ref:`license-discussion`.
 
-Copyright Policy
+Copyright policy
```
Comment (QuLogic) [inline doc/users/license.rst:15]: Possibly a proper noun, but not sure.

Comment (timhoffm) [reply doc/users/license.rst:15]: Seems to be used lowercase in plain text, e.g.

https://www.rightsdirect.com/10-tips-creating-copyright-policy/
http://www.copyright.com/learn/6-tips-creating-effective-copyright-policy/

File: doc/users/github_stats.rst (hunk 2 above)
Comment (timhoffm) [reply doc/users/github_stats.rst:6]: Changed the generator as well.

File: tools/github_stats.py [hunk 4]
```diff
@@ -176,13 +176,13 @@ def report(issues, show_urls=False):
     # Print summary report we can directly include into release notes.
     print('.. _github-stats:')
     print()
-    print('GitHub Stats')
+    print('GitHub statistics')
     print('============')
```
Comment (jklymak) [inline tools/github_stats.py:180]: These lengths need to match?
```suggestion
    print('=================')
```

**Review by jklymak — APPROVED** [review]

**Comment by timhoffm** [comment]
As this is a documentation change, I'm merging based on one approval.

**Pull request merged**